schema.graphql

```diff
@@ -107,6 +121,162 @@ type UnprocessedEntity @entity {
   block: Block
 }
 
+type Service @entity {
+  id: ID!
+  name: String!
+  computeUnitsPerRelay: BigInt!
+  owner: Account!
+  addServiceMsgs: [AddServiceMsg] @derivedFrom(field: "service")
+}
+
+type AppStakeMsg @entity {
+  id: ID!
+  stake: Coin!
+  transaction: Transaction!
+  block: Block!
+  application: Application! @index
+}
+
+type AppStakeMsgService @entity {
+  id: ID!
+  appStakeMsg: AppStakeMsg!
+  service: Service!
+}
+
+type AppUnstakeMsg @entity {
+  id: ID!
+  application: Application! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type DelegateToGatewayMsg @entity {
+  id: ID!
+  application: Application! @index
+  gateway: Gateway! @index
```

Comment:
(message relation) - I think you get it, I'm not going to comment on all of them.